test/workers/send_user_invitation_worker_test.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class SendUserInvitationWorkerTest < ActiveJob::TestCase
+  def test_provider_mailer_perform
+    Timecop.freeze(Time.zone.parse('2017-11-23 03:25:08 UTC +00:00')) do
+      ProviderInvitationMailer.any_instance.expects(:invitation).once
+
+      provider_account = FactoryBot.create(:provider_account)
+      invitation = FactoryBot.create(:invitation, account: provider_account)
+
+      assert_nil invitation.sent_at
+      SendUserInvitationWorker.new.perform(invitation.id)
+      assert invitation.reload.sent_at, Time.zone.now
+    end
+  end
+
+  def test_buyer_mailer_perform
+    Timecop.freeze(Time.zone.parse('2017-11-23 03:25:08 UTC +00:00')) do
+      InvitationMailer.any_instance.expects(:invitation).once
+
+      buyer_account = FactoryBot.create(:buyer_account)
+      invitation = FactoryBot.create(:invitation, account: buyer_account)
+
+      assert_nil invitation.sent_at
+      SendUserInvitationWorker.new.perform(invitation.id)
+      assert invitation.reload.sent_at
+    end
+  end
+
+  def test_handles_ssl_error
+    ProviderInvitationMailer.any_instance.expects(:invitation).raises(OpenSSL::SSL::SSLError)
+
+    invitation = FactoryBot.create(:invitation)
+
+    assert_nil invitation.sent_at
+    SendUserInvitationWorker.new.perform(invitation.id)
+    assert_nil invitation.reload.sent_at
+  end
+
+  def test_handles_smpt_error
+    ProviderInvitationMailer.any_instance.expects(:invitation).raises(Net::SMTPAuthenticationError)
+
+    invitation = FactoryBot.create(:invitation)
+
+    assert_nil invitation.sent_at
+    SendUserInvitationWorker.new.perform(invitation.id)
+    assert_nil invitation.reload.sent_at
+  end
+
+  def test_handles_socket_error
+    ProviderInvitationMailer.any_instance.expects(:invitation).raises(SocketError)
+
+    invitation = FactoryBot.create(:invitation)
+
+    assert_nil invitation.sent_at
+    SendUserInvitationWorker.new.perform(invitation.id)
+    assert_nil invitation.reload.sent_at
+  end
+
+  def test_handles_not_found_error
+    ProviderInvitationMailer.any_instance.expects(:invitation).raises(ActiveRecord::RecordNotFound)
```

Comment:
I think these tests could be just one. You can loop through the exceptions classes and repeat the same block. Just make sure to include another argument to the assertion where you pass the name of the exception class that failed, so we have useful message in case the test starts failing.